dali/pipeline/data/tensor_vector.h

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_DATA_TENSOR_VECTOR_H_
+#define DALI_PIPELINE_DATA_TENSOR_VECTOR_H_
+
+#include <memory>
+#include <vector>
+
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/data/tensor.h"
+
+#include "dali/kernels/tensor_shape.h"
+
+namespace dali {
+
+/**
+ * @brief Merges TensorList<Backend> and std::vector<std::shared_ptr<Tensor<Backend>>> APIs
+ * providing an uniform way of handling a collection/batch of tensors_.
+ *
+ * Propagates Buffer calls to every tensor uniformly
+ *
+ * @tparam Backend
+ */
+template <typename Backend>
+class TensorVector {
+ public:
+  TensorVector() {}
+  explicit TensorVector(int batch_size) : tensors_(batch_size, nullptr) {
+    for (auto &t : tensors_) {
+      t = std::make_shared<Tensor<Backend>>();
+    }
+  }
+
+  TensorVector(const TensorVector &) = delete;
```

Comment:
Why?

Reply:
Cause all of the others (Buffer/Tensor/TL) do that and I am not sure if we want to pass around copies of vectors of shared_ptrs.